pkg/scheduler/predicates/ha.go

```diff
@@ -127,13 +142,69 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
 	return getNodeFromNames(nodes, minNodeNames), nil
 }
 
+func (h *ha) realAcquireLock(pod *apiv1.Pod) (*apiv1.PersistentVolumeClaim, *apiv1.PersistentVolumeClaim, error) {
+	ns := pod.GetNamespace()
+	component := pod.Labels[label.ComponentLabelKey]
+	instanceName := pod.Labels[label.InstanceLabelKey]
+	podName := pod.GetName()
+	pvcList, err := h.pvcListFn(ns, instanceName, component)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currentPVCName := pvcName(component, podName)
+	var currentPVC *apiv1.PersistentVolumeClaim
+	var schedulingPVC *apiv1.PersistentVolumeClaim
+	for i := 0; i < len(pvcList.Items); i++ {
```

Comment:
```suggestion
	for _, pvc := range pvcList.Items {
```

Reply:
Use this style to the golang [common mistakes](https://github.com/golang/go/wiki/CommonMistakes)

Reply:
There is not use closures running as goroutines，If there is such a scene, then the way to use range can also avoid this problem.

Reply:
`currentPVC ` and `schedulingPVC ` are pointers,  the `pvc` in the for range block is a variable,    if we use the for range style, then these two variables will get the same address.

Reply:
For example:
```
package main

import "fmt"

func main() {
	arr := []string{"a", "b", "c"}

	var pa *string
	var pb *string
	for _, item := range arr {
		if item == "a" {
			pa = &item
		}
		if item == "b" {
			pb = &item
		}
	}
	fmt.Println(pa)
	fmt.Println(pb)
	fmt.Println(pa == pb)

	var ppa *string
	var ppb *string
	for i := 0; i < len(arr); i++ {
		if arr[i] == "a" {
			ppa = &arr[i]
		}
		if arr[i] == "b" {
			ppb = &arr[i]
		}
	}
	fmt.Println(ppa)
	fmt.Println(ppb)
	fmt.Println(ppa == ppb)
}
```

Outputs:
```
0xc00005e1c0
0xc00005e1c0
true
0xc000060180
0xc000060190
false
```

Reply:
You can create a new variable to fix this problem. such as follow:
```
for i := 0; i < len(pvcList.Items); i++ { 
    pvc := pvcList.Items[i]
    ...
}
------
for _, pvc := range pvcList.Items {
    pvc := pvc
 ...
}
```

Reply:
Style `for i:=0; i<n; i++` is equivalent to the `for range` style, but using `for i:=0; i<n; i++` is more clear at this situation.